microcosm_elasticsearch/searching.py

```diff
@@ -23,18 +25,45 @@ class SearchIndex:
         same type.
 
     """
-    def __init__(self, graph, index, model_class, doc_type=None):
+    __mapping_type_name__ = "doc"
+
+    @property
+    def mapping_type_name(self):
+        return self.__class__.__mapping_type_name__
+
+    @property
+    def doc_type_field(self):
+        """
+        Defines document field used to determine the document's polymorphic type in a single-mapping-type index
+
+        """
+        return "doctype"
+
+    def __init__(self, graph, index, doc_type=None):
         """
         :param graph: the object graph
         :param index: the name of an index to use
-        :param model_class: the model to return in the results list
         :param doc_type: the doc type to search for; uses the index's doc types if omitted
 
         """
         self.elasticsearch_client = graph.elasticsearch_client
         self.index = index
-        self.model_class = model_class
-        self.doc_type = doc_type
+        # Mapping from ES custom type field to corresponding model class
+        self.doc_types = dict()
+        self.mapping = Mapping(self.mapping_type_name)
+        if doc_type is not None:
+            self.register_doc_type(doc_type)
```

Comment:
The key change in `SearchIndex` is that we have to define an index `mapping` ourselves. This is because `elasticsearch_dsl` still ties one mapping to one `DocType`; if we define several `Model`s (or `DocType`s)  with the same mapping type (e.g. both indexed under `/index_name/doc/`), then the one that is declared last will override the mapping, and the fields of the first models won't show up in the index mapping.

Reply:
I thought the plan was to define a single mapping for a single doc type? That the mapping would be an amalgamation of all the different Models. Or is this to specify which fields are returned from the query?